projects/make-a-vis/src/lib/data-view/shared/data.service.ts

```diff
@@ -50,18 +71,27 @@ export class DataService {
             dataSource.hiddenData = false;
             dataSource.numRows = 0;
             dataSource.streamId = get(recordSet.defaultRecordStream, 'id');
+
             const operator = this.getDataMappingOperator(recordSet.dataVariables, project.graphicVariables, recordSet.id);
             if (recordSet.defaultRecordStream) {
-              recordSet.defaultRecordStream.asObservable().subscribe((changeSet: RawChangeSet<any>) => {
-                dataSource.data = (changeSet.insert || []).map(operator.getter);
-                dataSource.numRows += (changeSet.insert.length || 0) - (changeSet.remove.length || 0);
-                if (changeSet.insert.length > 1) {
-                  dataSource.label = recordSet.labelPlural;
-                }
-              });
+              const data: any[] = [];
```

Comment:
Logic is wrong here. this data array is never updated. Works with static data, but will not work with streaming data. You should be saving the results of this.updateData back into data.